src/stages/main/patchers/WhilePatcher.js

```diff
@@ -125,9 +126,16 @@ export default class WhilePatcher extends NodePatcher {
   patchAsExpression() {
     this.body.setImplicitlyReturns();
     let resultBinding = this.getResultArrayBinding();
-    this.insert(this.contentStart, `(() => { ${resultBinding} = []; `);
     this.patchAsStatement();
-    this.insert(this.contentEnd, ` return ${resultBinding}; })()`);
+    let prefix = this.yielding ? 'yield* (function*()' : '(() =>';
+    this.insert(this.contentStart, `${prefix} { ${resultBinding} = []; `);
+    let postfix = this.yielding ? '.call(this)' : '()';
```

Comment:
Coffee itself passes arguments _only_ when referenced inside body. So, if we support this, we might go for the same logic? Isn't it a rare case and produces unreadable code if arguments are not referenced? Also, it was not present in the logic before this PR and it disallows the nice and readable usage of the fat arrow. Would suggest to support this at a later time?